url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FldExporter.java

```diff
@@ -0,0 +1,214 @@
+/*   Copyright 2013 Juan Rada-Vilela
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.fuzzylite.imex;
+
+import com.fuzzylite.Engine;
+import com.fuzzylite.Op;
+import com.fuzzylite.variable.InputVariable;
+import com.fuzzylite.variable.OutputVariable;
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ *
+ * @author jcrada
+ */
+public class FldExporter extends Exporter {
+
+    public static final int DEFAULT_MAXIMUM = 1024;
+    public static final String DEFAULT_SEPARATOR = " ";
+
+    protected String separator;
+
+    public FldExporter() {
+        this(DEFAULT_SEPARATOR);
+    }
+
+    public FldExporter(String separator) {
+        this.separator = separator;
+    }
+
+    public String getSeparator() {
+        return separator;
+    }
+
+    public void setSeparator(String separator) {
+        this.separator = separator;
+    }
+
+    public String header(Engine engine) {
+        String inputsHeader = headerInputVariables(engine.getInputVariables());
+        String outputsHeader = headerOutputVariables(engine.getOutputVariables());
+        StringBuilder result = new StringBuilder();
+        result.append(String.format("@Engine: %s;", engine.getName()));
+        if (!inputsHeader.isEmpty()) {
+            result.append(separator).append(inputsHeader);
+        }
+        if (!outputsHeader.isEmpty()) {
+            result.append(separator).append(outputsHeader);
+        }
+        return result.toString();
+    }
+
+    public String headerInputVariables(List<InputVariable> inputVariables) {
+        List<String> result = new ArrayList<>();
+        for (InputVariable inputVariable : inputVariables) {
+            if (inputVariable.isEnabled()) {
+                result.add("@InputVariable: " + inputVariable.getName() + ";");
+            }
+        }
+        return Op.join(result, separator);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 71:77, 81:87.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.